### PR TITLE
Fixed #31136 -- Disabled grouping by aliases on QuerySet.values()/values_list().

### PR DESCRIPTION
**Pull request by felixxm**
Regression in fb3f034f1c63160c0ff13c609acd01c18be12f80.

Thanks Sigurd Ljødal for the report.

ticket-31136

### CODE REVIEW

File: tests/aggregation/tests.py [hunk 1]
```diff
@@ -1151,6 +1151,14 @@ def test_aggregation_subquery_annotation_exists(self):
         )
         self.assertTrue(publisher_qs.exists())
 
+    def test_aggregation_exists_annotation(self):
+        published_books = Book.objects.filter(publisher=OuterRef('pk'))
+        publisher_qs = Publisher.objects.annotate(
+            published_book=Exists(published_books),
+            count=Count('book'),
+        ).values_list('name', 'published_book')
+        self.assertTrue(publisher_qs.exists())
```
Comment (atombrella) [inline tests/aggregation/tests.py]: Per new code guidelines, can we use `assertIs`? :)

Comment (felixxm) [reply tests/aggregation/tests.py]: TBH I added `.exists()` only to evaluate queryset, this issue is not related with `exists()`, so I will change this assertion.

Comment (felixxm) [reply tests/aggregation/tests.py]: Updated.

**Comment by felixxm** [comment]
buildbot, test on oracle.

**Pull request merged**